### DISCUSSION TITLE
useResponseCache (TTL)

### DISCUSSION THREAD

**Discussion by tylerjbainbridge** [dotansimha/envelop#2158] [Q&A]
Hi all,

I'm using `useResponseCache` for a project and have a top level default TTL set to 30000, but I also have `scopePerSchemaCoordinate`, `ttlPerType`, and `ttlPerSchemaCoordinate` set for specific instances.

It seems like all of the keys being added to redis have no TTL set (infinite) which I can't seem to track down. 

Any idea why this may be happening?

Tyler

**Comment by EmrysMyrddin** [comment]
Hi, thank you for reaching out!
Can you share your setup or provide a reproduction case?

Which version are you using ? We used to have a bug leading to infinite TTL in some cases but it should be fixed now.

**Comment by tylerjbainbridge** [comment]
Hi! Here are my versions.

My set up is Apollo Server & Envelop, and I'm using redis with the response cache.

Things are ending up in Redis as intended, but everything has an infinite TTL for some reason.

 "@envelop/core": "^5.0.0",
  "@envelop/response-cache": "^6.1.2",
  "@envelop/response-cache-redis": "^4.1.2",

```js
     useResponseCache({
            cache: graphqlCache,
            includeExtensionMetadata: true,
            // Default TTL 10 seconds.
            ttl: 30000,
            scopePerSchemaCoordinate: {
             ...
            },
            ttlPerType: {
              ...
            },
            ttlPerSchemaCoordinate: {
              ...
            },
          }),
```

**Reply by EmrysMyrddin** [reply]
Hi, sorry to have dropped the ball here, can you open an issue? It will be easier to track than in discutions.
